contextlib2/__init__.pyi

```diff
@@ -121,12 +121,15 @@ if True:
             __traceback: Optional[TracebackType],
         ) -> Awaitable[bool]: ...
 
+
 if True:
-    class nullcontext(AbstractContextManager[_T]):
+    class nullcontext(AbstractContextManager[_T], AbstractAsyncContextManager[_T]):
         enter_result: _T
         @overload
         def __init__(self: nullcontext[None], enter_result: None = ...) -> None: ...
         @overload
         def __init__(self: nullcontext[_T], enter_result: _T) -> None: ...
         def __enter__(self) -> _T: ...
-        def __exit__(self, *exctype: Any) -> bool: ...
+        def __exit__(self, *exctype: Any) -> None: ...
+        async def __aenter__(self) -> _T: ...
```

Comment:
@hauntsaninja do you know why stubcheck didn't catch these missing methods?

Reply:
so there are two things that went wrong: 1) an intentional-for-now stubtest decision, 2) a feature that's missing in stubtest on master.

1. in figuring out what definitions of a class to check, stubtest will always check [all definitions in the stub](https://github.com/python/mypy/blob/198592832b2223122090e00f6bcac9e1e24d6018/mypy/stubtest.py#L240). but for things missing from the stub, stubtest will only [check non private definitions and an allowlisted set of dunders on the runtime](https://github.com/python/mypy/blob/198592832b2223122090e00f6bcac9e1e24d6018/mypy/stubtest.py#L245). [this comment](https://github.com/python/mypy/blob/198592832b2223122090e00f6bcac9e1e24d6018/mypy/stubtest.py#L241) mentions why i did this, but this was a bad experience for you so i'll revisit this logic.
2. stubtest on master doesn't check base classes, which could also have potentially caught this issue. i have [a branch](https://github.com/python/typeshed/issues/3968) that does this, but i don't love that implementation. this is probably the biggest hole in stubtest on master.

Reply:
Thanks for the thorough explanation! I'm glad it's not just a config problem on my end